lib/puppet/type/route53_spf_record.rb

```diff
@@ -0,0 +1,15 @@
+require_relative '../../puppet_x/puppetlabs/route53_record'
+
+Puppet::Type.newtype(:route53_spf_record) do
+  extend PuppetX::Puppetlabs::Route53Record
+  @doc = 'Type representing a Route53 SPF record.'
+  create_properties_and_params()
+
+  # SPF records should always be wrapped in double quotes
+  # this munge avoids the need to pass in a '"value"' to Puppet
+  values_property = self.properties.find { |item| item == Puppet::Type::Route53_spf_record::Values }
+  values_property.munge do |value|
+    value =~ /^".+"$/ ? value : "\"#{value}\""
```

Comment:
The rationale here was that if you know DNS, then providing the quotes is more natural. If you don't know DNS it's the opposite. This also mirrors the behaviour of the AWS console.

My concern with choosing one is that without quotes is sort of wrong - the record will actually have quotes. But with quotes it's also unexpected if you don't know that. This felt like the nicest interface for different users.